File: frontend/src/types/model_config.ts
export interface ModelConfigResponse {
  id: string
  name: string
  provider: string
  model_name: string
  base_url: string | null
  category: "llm" | "embedding" | "vision"
  role: "general" | "fast" | null
  temperature: number | null
  max_output_tokens: number | null
  context_size: number | null
  is_default: boolean
  is_active: boolean
  json_mode_enabled: boolean
  supports_vision: boolean
  created_at: string
  updated_at: string | null
  // api_key is never returned from the backend
}

export interface ModelConfigCreate {
  name: string
  provider: string
  model_name: string
  base_url?: string | null
  api_key?: string | null
  category?: "llm" | "embedding" | "vision"
  role?: "general" | "fast" | null
  temperature?: number | null
  max_output_tokens?: number | null
  context_size?: number | null
  is_default?: boolean
  json_mode_enabled?: boolean
  supports_vision?: boolean
}

export type ModelConfigUpdate = Partial<ModelConfigCreate & { is_active: boolean; json_mode_enabled: boolean; supports_vision: boolean }>

File: src/fim_one/web/schemas/model_config.py
"""Model configuration request/response schemas."""

from __future__ import annotations

from typing import Literal

from pydantic import BaseModel, Field


class ModelConfigCreate(BaseModel):
    name: str = Field(min_length=1, max_length=100)
    provider: str = ""
    model_name: str
    base_url: str | None = None
    api_key: str | None = None
    category: Literal["llm", "embedding", "vision"] = "llm"
    temperature: float | None = None
    max_output_tokens: int | None = None
    context_size: int | None = None
    is_default: bool = False
    role: Literal["general", "fast"] | None = None
    json_mode_enabled: bool = True
    supports_vision: bool = False


class ModelConfigUpdate(BaseModel):
    name: str | None = None
    provider: str | None = None
    model_name: str | None = None
    base_url: str | None = None
    api_key: str | None = None
    category: str | None = None
    temperature: float | None = None
    max_output_tokens: int | None = None
    context_size: int | None = None
    is_default: bool | None = None
    is_active: bool | None = None
    role: str | None = None
    json_mode_enabled: bool | None = None
    supports_vision: bool | None = None


class ModelConfigResponse(BaseModel):
    id: str
    name: str
    provider: str
    model_name: str
    base_url: str | None
    category: str
    temperature: float | None
    max_output_tokens: int | None
    context_size: int | None
    role: str | None
    is_default: bool
    is_active: bool
    json_mode_enabled: bool
    supports_vision: bool
    created_at: str
    updated_at: str | None
    # NEVER expose api_key in responses
